geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py

```diff
@@ -0,0 +1,152 @@
+                    with Workspace(filepath) as iter_workspace:
+
+                        receiver_entity = extract_dcip_survey(
+                            iter_workspace,
+                            data.entity,
+                            self.pseudo3d_params.line_object.values,
+                            trial["line_id"],
+                        )
+
+                        mesh = get_drape_model(
+                            iter_workspace,
+                            "Models",
+                            receiver_entity.vertices,
+                            [
+                                self.pseudo3d_params.u_cell_size,
+                                self.pseudo3d_params.v_cell_size,
+                            ],
+                            self.pseudo3d_params.depth_core,
+                            [self.pseudo3d_params.horizontal_padding] * 2
+                            + [self.pseudo3d_params.vertical_padding, 1],
+                            self.pseudo3d_params.expansion_factor,
+                        )[0]
+
+                        iter_workspace.remove_entity(receiver_entity.current_electrodes)
+                        iter_workspace.remove_entity(receiver_entity)
+
+                        xyz_out = mesh.centroids
+                        model_uids = deepcopy(models)
+                        for name, model in models.items():
+                            if model is None:
+                                continue
+                            elif isinstance(model, Data):
+                                model_values = weighted_average(
+                                    xyz_in, xyz_out, [model.values], n=1
+                                )[0]
+                            else:
+                                model_values = model * np.ones(len(xyz_out))
+
+                            model_object = mesh.add_data(
+                                {name: {"values": model_values}}
+                            )
+                            model_uids[name] = model_object.uid
+
+                        for key in ifile.data:
+                            param = getattr(self.pseudo3d_params, key, None)
+                            if hasattr(param, "uid"):
+                                if not isinstance(param, Data):
+                                    param.copy(
+                                        parent=iter_workspace, copy_children=True
+                                    )
+
+                            ifile.data[key] = param
+
+                        ifile.data.update(
+                            dict(
+                                lookup[uuid],
+                                **{
+                                    "title": ifile.data["title"].replace("batch ", ""),
+                                    "inversion_type": ifile.data[
+                                        "inversion_type"
+                                    ].replace("pseudo 3d", "2d"),
```

Comment:
ifile is already 2D, don't think we need any of the text replacements.

Reply:
I ended up with the title and inversion_type of the batch 2d from the self.pseudo_params update loop.  I've bypassed those to params within the loop rather than correcting after